scripts/build_native_deps.sh

```diff
@@ -295,7 +295,15 @@ build_guile()
 
     extract "$GUILE_ARCHIVE" "$src"
     # Export dynamic symbols from guile executable so that srfi modules work.
-    sed_i "s|guile_LDFLAGS = .*$|& -Wl,--export-dynamic|" "$src/libguile/Makefile.in"
+    sed_i "s|guile_LDFLAGS = .*$|& $LDFLAGS|" "$src/libguile/Makefile.in"
+
+    if [ "$uname" = "Darwin" ]; then
+        export WHOLE_ARCHIVE="-Wl,-all_load -Wl,-undefined,dynamic_lookup"
+        export NO_WHOLE_ARCHIVE=""
+    else
+        export WHOLE_ARCHIVE="-Wl,--whole-archive"
+        export NO_WHOLE_ARCHIVE="-Wl,--no-whole-archive"
+    fi
```

Comment:
I don't think we need to export them. Can you put this into `scripts/common.sh`, next to `LDFLAGS_EXPORT_DYNAMIC`?

Reply:
Sure!